### DISCUSSION TITLE
Feature Request: Support Data Import from Avro Files

### DISCUSSION THREAD

**Discussion by abhioncbr** [duckdb/duckdb#7085] [Ideas]
Why do you want this feature?
Avro is one of the most popular and widely used formats for exchanging big data. Like Parquet, it is also heavily used, especially for data exchange in/from Kafka. I think as duckdb getting more popular day by day, it should support data import based on Avro files.

[a comment by an automated account is not shown]

**Comment by hannes** [comment]
I started working on this. It would help if the people upvoting this could email me sample Avro files to hannes@duckdblabs.com - please only include files that we can add as test cases to the public repo

**Comment by Mytherin** [comment]
The [community extension that can read Avro is now available](https://duckdb.org/community_extensions/extensions/avro.html) (developed by @hannes). See also the [blog post](https://duckdb.org/2024/12/09/duckdb-avro-extension.html). 

The extension lives in [hannes/duckdb_avro](https://github.com/hannes/duckdb_avro), issues/suggestions can be filed there.

**Reply by erovira** [reply]
> The extension does not contain Avro write functionality. This is on purpose, by not providing a writer we hope to decrease the amount of Avro files in the world over time

I'm curious about this statement in the linked blogpost. @hannes would you mind elaborating a bit on it? Sorry if you have already and I missed it.

And needless to say: thank you for your work on duckdb and in particular for this extension!
